comparision-plugin.js

```diff
@@ -1,6 +1,45 @@
 'use strict';
 
 const semver = require('semver');
+const NPMDependencyChecker = require('ember-cli-version-checker/src/npm-dependency-version-checker');
+const extractTrueVersion = require('./utils/extract-true-version');
+
+function versionFor(depName, state) {
+  let VERSIONS = state.version_cache;
+  let version = VERSIONS[depName];
+
+  if (version === undefined) {
+    let checker = new NPMDependencyChecker(
+      {
+        _addon: {
```

Comment:
you shouldn't need to mock this `_addon` property, I would think you can do `new VersionChecker({ project: { root: state.opts.root } })`

Reply:
@rwjblue no. the dependency checkers take the `name` and `root` values from the `_addon` field unfortunately.

Reply:
root: https://github.com/ember-cli/ember-cli-version-checker/blob/master/src/npm-dependency-version-checker.js#L10

name: https://github.com/ember-cli/ember-cli-version-checker/blob/master/src/dependency-version-checker.js#L54

Reply:
Right, thats why we should go through the main entry (see above) 😉 

https://github.com/ember-cli/ember-cli-version-checker/blob/master/src/version-checker.js#L18

Reply:
resolved by using VersionChecker